vector/src/main/scala/geotrellis/vector/SpatialIndex.scala

```diff
@@ -88,6 +91,62 @@ class SpatialIndex[T](val measure: Measure = Measure.Euclidean) extends Serializ
 
   def pointsInExtentAsJavaList(extent: Extent): java.util.List[T] =
     rtree.query(new Envelope(extent.xmin, extent.xmax, extent.ymin, extent.ymax)).asInstanceOf[java.util.List[T]]
+
+  def kNearest(x: Double, y: Double, k: Int): List[T] =
+    kNearest(new Envelope(new Coordinate(x, y)), k)
+
+  def kNearest(pt: (Double, Double), k: Int): List[T] =
+    kNearest(new Envelope(new Coordinate(pt._1, pt._2)), k)
+
+  def kNearest(ex: Extent, k: Int): List[T] = {
+    case class PQitem[A](d: Double, x: A) extends Ordered[PQitem[A]] {
+      import scala.math.Ordered.orderingToOrdered
+      def compare(that: PQitem[A]): Int = { (this.d) compare (that.d) }
+    }
+
+    val gf = new GeometryFactory()
+    val env = ex.jtsEnvelope
+    val pq = (new mutable.PriorityQueue(): mutable.PriorityQueue[PQitem[AbstractNode]]).reverse
+    val best = (new mutable.PriorityQueue(): mutable.PriorityQueue[PQitem[T]])
+
+    def addToBest (elem: PQitem[T]) = {
+      if (best.size < k)
+        best.enqueue(elem)
+      else {
+        if (elem.d < best.head.d) {
+          best.enqueue(elem)
+          best.dequeue()
+        }
+      }
+    }
+    def ibToPQitem (ib: ItemBoundable): PQitem[T] = {
+      PQitem(DistanceOp.distance(gf.toGeometry(env), gf.toGeometry(ib.getBounds.asInstanceOf[Envelope])), ib.getItem.asInstanceOf[T])
+    }
+    def anToPQitem (nd: AbstractNode): PQitem[AbstractNode] = {
```

Comment:
These names are really hard to follow. I'd prefer more verbose names that explained a bit more what they are doing than things like `ib` and `an`